### ISSUE TITLE
Jitsi and Java 9

### ISSUE THREAD

**Issue by ArthurBorsboom** [jitsi/jitsi#403]
Java 9 has been released September 22, 2017. While changing the default on my system to test which application would need updating, Jitsi seems to be one of them.

Several warnings and errors are generated on the CLI when starting Jitsi, such as

`WARNING: Illegal reflective access by org.apache.felix.framework.util.SecureAction (file:/usr/lib/jitsi/lib/felix.jar) to method java.net.URLClassLoader.addURL(java.net.URL)`

`org.osgi.framework.BundleException: Unable to resolve org.jitsi.bccontrib`

`org.osgi.framework.BundleException: Unable to resolve slf4j.api`

A full list of errors and warnings on my system (Arch Linux and Jitsi 2.11.20170821): https://pastebin.com/v79cnGN6

**Comment by ibauersachs** [comment]
The reflective access stuff can likely be fixed with the `--illegal-access=permit` JVM flag. But I assume we need to update Apache Felix too.

**Comment by ArthurBorsboom** [comment]
Hereby I share my two findings.

Jitsi (git) does not compile with Java 9.
Jitsi (git) does compile with Java 8 and then runs with Java 9, with the following reflective warning(s).

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.felix.framework.util.SecureAction (file:/usr/lib/jitsi/lib/felix.jar) to method java.net.URLClassLoader.addURL(java.net.URL)
WARNING: Please consider reporting this to the maintainers of org.apache.felix.framework.util.SecureAction
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

Is this what you expected?

**Comment by ibauersachs** [comment]
Yes.

The compile issues are much more difficult to solve, e.g. the nameservice SPI has been dropped; it is thus now impossible to guarantee that DNS lookup don't bypass DNSSEC (if enabled). Not sure without actually trying to compile what else fails.

The warnings you quoted are from Apache Felix, the OSGi framework we use. Feel free to open an issue there.

Right now my hands are tied with the Smack 4.2 migration (which largely impacts Jitsi Meet). Once that's done, I plan to do other cleanups here. Some help wouldn't hurt...

**Comment by ullashunka** [comment]
I am slightly new ant java but I believe using new felix http://www.java2s.com/Code/Jar/f/Downloadfelixjar.htm can resolve nameservice problem

**Comment by ibauersachs** [comment]
If you're talking about the DNS nameservice, then no. It has been removed from the JRE and no library can replace a service that isn't called _from_ classes of the JRE anymore.

If you're talking about the other Felix issue with the boot classloader, this is fixed in our repo and will be fixed in Felix 5.6.10 (which AFAIK isn't released yet).
